### PR TITLE
added parameter to enable/disable saml authorization

### LINKED ISSUE #1520: Add possibility to prevent saml auth from being initialized

### PR DESCRIPTION
**Pull request by ssuvorov-fls**
fixes #1520

### CODE REVIEW

**Comment by chrisknoll** [comment]
Hello, @ssuvorov-fls ,
  I think this was touched in this PR: https://github.com/OHDSI/WebAPI/pull/1524, but that PR only tries to handle misconfigured SAML, not enable/disable.  This PR has a conflict, but probably related to the above PR.

Could you rebase this PR onto the latest master, and decide if you want to incorporate the exception handling in addition to enable/disable of SAML?

**Comment by ssuvorov-fls** [comment]
Master was merger into branch
I don't think that exception handling should be used here because we can miss such exceptions during startup

**Comment by chrisknoll** [comment]
Ok, go ahead and take it out. The application wasn't able to even start but if we can disable saml to prevent the exception, then that's another way.  I'd personally prefer it if saml did fail to initialize, the app could continue to start up, but we could report something in the `/info` endpoint about saml being disabled, allowing people to check if SAML is working or not without having to go into logs to try to understand why WebAPI failed to start.

File: src/main/java/org/ohdsi/webapi/shiro/management/AtlasRegularSecurity.java [hunk 1]
```diff
@@ -344,10 +347,11 @@ protected FilterChainBuilder getFilterChainBuilder() {
 
         if (this.samlEnabled) {
             filterChainBuilder
-                .addPath("/user/login/saml", SSL, CORS, FORCE_SESSION_CREATION, SAML_AUTHC, UPDATE_TOKEN, SEND_TOKEN_IN_URL)
-                .addPath("/user/saml/callback", SSL, HANDLE_SAML, UPDATE_TOKEN, SEND_TOKEN_IN_URL);
+                    .addPath("/user/login/saml", SSL, CORS, FORCE_SESSION_CREATION, SAML_AUTHC, UPDATE_TOKEN, SEND_TOKEN_IN_URL);
+            filterChainBuilder
+                    .addPath("/user/saml/callback", SSL, HANDLE_SAML, UPDATE_TOKEN, SEND_TOKEN_IN_URL);
```
Comment (YaroslavTir) [inline src/main/java/org/ohdsi/webapi/shiro/management/AtlasRegularSecurity.java:352]: can be done as one operation:
```
            filterChainBuilder
                    .addPath("/user/login/saml", SSL, CORS, FORCE_SESSION_CREATION, SAML_AUTHC, UPDATE_TOKEN, SEND_TOKEN_IN_URL)
                    .addPath("/user/saml/callback", SSL, HANDLE_SAML, UPDATE_TOKEN, SEND_TOKEN_IN_URL);
```

Comment (ssuvorov-fls) [reply src/main/java/org/ohdsi/webapi/shiro/management/AtlasRegularSecurity.java:351]: fixed

File: src/main/java/org/ohdsi/webapi/shiro/management/AtlasRegularSecurity.java [hunk 2]
```diff
@@ -240,8 +240,9 @@
 
     @Value("${security.cas.casticket}")
     private String casticket;
-    
-    private boolean samlEnabled = false;
+
+    @Value("${security.saml.emabled:false}")
```
[the remark this answers is not in the document]
Comment (ssuvorov-fls) [reply src/main/java/org/ohdsi/webapi/shiro/management/AtlasRegularSecurity.java]: fixed

File: pom.xml [hunk 3]
```diff
@@ -134,6 +134,7 @@
     <security.duration.initial>10</security.duration.initial>
     <security.duration.increment>10</security.duration.increment>
 
+    <security.saml.emabled>false</security.saml.emabled>
```
[the remark this answers is not in the document]
Comment (ssuvorov-fls) [reply pom.xml]: fixed

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -151,6 +151,7 @@ security.ad.userMapping.middlenameAttr=${security.ad.userMapping.middlenameAttr}
 security.ad.userMapping.lastnameAttr=${security.ad.userMapping.lastnameAttr}
 security.ad.userMapping.usernameAttr=${security.ad.userMapping.usernameAttr}
 
+security.saml.emabled=${security.saml.emabled}
```
[the remark this answers is not in the document]
Comment (ssuvorov-fls) [reply src/main/resources/application.properties]: fixed

**Pull request merged**